cypress/support/bot.ts

```diff
@@ -32,32 +38,51 @@ declare global {
              * @param displayName the display name to give to the bot user
              */
             getBot(synapse: SynapseInstance, displayName?: string): Chainable<MatrixClient>;
+            /**
+             * Create a new Matrix client to interact with the API as user
+             * separate from the one logged in
+             * @param synapse the instance on which to register the bot user
+             * @param opts Options to pass when creating a new Matrix client
+             *     like `userId` and `accessToken`
+             */
+            newMatrixClient(synapse: SynapseInstance, opts: INewMatrixClientOptions): Chainable<MatrixClient>;
         }
     }
 }
 
+Cypress.Commands.add("newMatrixClient", (
+    synapse: SynapseInstance,
+    { userId, accessToken, deviceId }: INewMatrixClientOptions,
+): Chainable<MatrixClient> => {
+    return cy.window({ log: false }).then(win => {
+        const cli = new win.matrixcs.MatrixClient({
+            baseUrl: synapse.baseUrl,
+            userId,
+            deviceId,
+            accessToken,
+            request,
+        });
+
+        cli.on(win.matrixcs.RoomMemberEvent.Membership, (event, member) => {
+            if (member.membership === "invite" && member.userId === cli.getUserId()) {
+                cli.joinRoom(member.roomId);
+            }
+        });
```

Comment:
This behaviour is unexpected for this method

Reply:
I can separate it back to just `getBot` 👍

Didn't seem too unreasonable to have it here (respond to invite by joining the room) but could see a situation where you wouldn't want it to happen. Same could kinda be argued for `getBot` but `newMatrixClient` is a lot more raw 🤔